src/log.ts

```diff
@@ -866,6 +870,49 @@ class Log implements LogSeverityFunctions {
     });
   }
 
+  /**
+   * Truncate log entries at maxEntrySize, so that error is not thrown, see:
+   * https://cloud.google.com/logging/quotas
+   *
+   * @private
+   *
+   * @param {object|string} the JSON log entry.
+   * @returns {object|string} truncated JSON log entry.
+   */
+  private truncateEntries(entries: EntryJson[]) {
+    return entries.forEach(entry => {
+      if (this.maxEntrySize === undefined) return;
+
+      const payloadSize = JSON.stringify(entry).length;
+      if (payloadSize < this.maxEntrySize) return;
+
+      const delta = payloadSize - this.maxEntrySize;
+      if (entry.textPayload) {
+        entry.textPayload = entry.textPayload.slice(
+          0,
+          Math.max(entry.textPayload.length - delta, 0)
+        );
+      } else {
+        // Stackdriver Log Viewer picks up the summary line from the
+        // 'message' field.
+        let text: string | null | undefined;
+        if (
+          entry.jsonPayload &&
+          entry.jsonPayload.fields &&
+          entry.jsonPayload.fields.message
+        ) {
+          text = entry.jsonPayload.fields.message.stringValue;
+        }
+        if (text) {
+          entry!.jsonPayload!.fields!.message.stringValue = text.slice(
```

Comment:
If you move this `if (text)` block and this assignment one line up, inside the previous `if`, you won't need to use `!` and the code might even become more straightforward.